workspaces/mcp-integrations/plugins/backstage-mcp-tools-backend/src/plugin.ts

```diff
@@ -250,31 +250,82 @@ export async function fetchCatalogEntities(
   if (input?.type) {
     filter['spec.type'] = input.type;
   }
+  if (input?.name) {
+    filter['metadata.name'] = input.name;
+  }
+  if (input?.owner) {
+    filter['spec.owner'] = input.owner;
+  }
+  if (input?.lifecycle) {
+    filter['spec.lifecycle'] = input.lifecycle;
+  }
+  if (input?.tags) {
+    filter['metadata.tags'] = input.tags;
+  }
 
-  const { items } = await catalog.getEntities(
-    {
-      fields: [
-        'metadata.name',
-        'kind',
-        'metadata.tags',
-        'metadata.description',
-        'spec.type',
-      ],
-      filter,
-    },
-    {
-      credentials,
-    },
+  const getEntitiesOptions: any = {
+    filter,
+  };
+
+  // When using the reduced output, we can also reduce the number of fields fetched via the API
+  if (!input?.verbose) {
+    getEntitiesOptions.fields = [
+      'metadata.name',
+      'kind',
+      'metadata.tags',
+      'metadata.description',
+      'spec.type',
+      'spec.owner',
+      'spec.lifecycle',
+      'relations',
+    ];
+  }
+
+  // Avoid potentially logging PII when we log which filters are being used
+  const loggedFilters = {
```

Comment:
I wanted to log the filters that were being used with the catalog API call, but was worried about potential PII in the entity owner and name fields (e.g. Users are treated as catalog entities).